gen3utils/etl/etl_validator.py

```diff
@@ -1,202 +1,310 @@
+                        src, json_obj
+                    )
+                )
+            )
+    return name
+
+
+def validate_path(
+    json_obj, grouping_path, recorded_errors, list_of_nodes, nodes_with_props
+):
+    path = json_obj.get("path", grouping_path)
+    names = []
+    if path is None:
+        recorded_errors.append(
+            PropertiesError(
+                "Missing path declaration for the property {}.".format(json_obj)
+            )
+        )
+    else:
+        path_items = path.split(".")
+        if "_ANY" in path_items:
+            path_items.remove("_ANY")
+        for item in path_items:
+            # get the edge name and the property definition out of the line:
+            # subjects[subject_id:id,project_id]
+            [edge, str_fields] = (
+                list(filter(None, re.split(r"[\[\]]", item)))
+                if item.find("[") != -1
+                else [item, None]
+            )
+            if edge not in list_of_nodes:
+                recorded_errors.append(PathError(path))
+            if str_fields is not None:
+                fields = str_fields.split(",")
+                for f in fields:
+                    src = f.split(":")[-1] if f.find(":") != -1 else f
+                    name = f.split(":")[0] if f.find(":") != -1 else f
+                    names.append(
+                        validate_name_src(
+                            {"name": name, "src": src},
+                            edge,
+                            recorded_errors,
+                            nodes_with_props,
+                        )
+                    )
+    return names
+
+
+def get_all_nodes(model):
+    labels_to_back_refs = {}
+    """
+    dictionary from label to back_ref
+    {
+        "subject": "subjects"
+    }
+    """
+    nodes_with_props = {}
+    """
+    dictionary nodes (with plural) to all properties
+    {
+        "subjects": [
+            "submitter_id",
+            "project_id",
+            "species"
+        ]
+    }
+    """
+    categories_to_labels = defaultdict(list)
+    """
+    group label by its category
+    {
+        "data_file": [
+            submitted_aligned_reads,
+            submitted_unaligned_reads
+        ]
+    }
+    """
+    all_classes = model.Node.get_subclasses()
+    for n in all_classes:
+        present_links = n._pg_edges
+        present_props = n.__pg_properties__
+        present_node_props = list(present_props.keys()) + ["id"]
+        category = n._dictionary.get("category")
+        categories_to_labels[category].append(n.label)
+        backref = ""
+        if len(present_links) > 0:
+            backref = list(present_links.values())[0].get("backref")
+            labels_to_back_refs[n.label] = backref
+        nodes_with_props.update({backref: present_node_props})
+    return labels_to_back_refs, nodes_with_props, categories_to_labels
+
+
+def validate_mapping(dictionary_url, mapping_file):
```

Comment:
would you mind adding a call to a function `check_mappings_format()` right after `def validate_mapping`, just as a reminder, and we can leave this function as a TODO